QQE-1584.md

```diff
@@ -0,0 +1,82 @@
+# QQE-1584 Quarkus LangChain4J testing
+
+JIRA link: https://issues.redhat.com/browse/QQE-1584
+
+This is a high-level strategy for testing of langchain4j
+
+## Scope of the testing
+@SystemMessage + @UserMessage annotations
+RAG functionality
+Integration with MCP API
+
+### Optional, but desirable:
+@Tool + @RegisterAiService (only supported for OpenAI, so cannot be tested locally)
+Observability for tool requests (to avoid a situation, where LLM overwhelms the database with too many requests)
+
+## Existing coverage upstream:
+### Workshop
+https://quarkus.io/quarkus-workshop-langchain4j/step-11/#running-the-llm-inference-locally  and  https://github.com/quarkusio/quarkus-workshop-langchain4j/tree/main/step-11
+(they are not run automatically, as far as I can tell)
+The tests use jlama, which requires Java 21+
+Attempt to run the test locally with jlama fails on my laptop(32.0 GiB RAM, 13th Gen Intel® Core™ i7-1365U × 12). Will try to run it on server, but I am not very optimistic about that.
+Successfully ran it on the server. The downside is there is no rest api, and we have to connect through websocket (which is not optimal). Also, it hangs regularly, even on xlarge.
+
+### Quakriverse extension:
+There are integration tests modules in the extension, but they contain no tests at the moment.
+According to Mario, this is intentional, since they consider upstream lang4chain tests to be enough (but we do not!)
+Ollama modules can be started on the laptop in dev mode. Jlama ones can be run as well.
+
+## Automated test development
+### Functional testing
+1. Add some tests for existing functionality into https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/ollama, https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/simple-ollama and https://github.com/quarkiverse/quarkus-langchain4j/tree/main/integration-tests/jlama
+2. Beware, that these modules use models of different size, and this can affect reproducibility of tests. Some tests may not work on our hardware with some of the models.
```

Comment:
How do we proceed if the tests don't work on our hardware? What makes our hardware specific and what is the strategy to address that?